First action not executed when using multiple member function actions

SML 1.1.3 introduced the option to specify member function pointer instead of using lambda.
The normal case as shown in the example works, but when specifying multiple actions for a single event, I get a compilation warning:
> [build] /opt/build/debug/src/test.cxx: In member function ‘auto {anonymous}::actions_guards::operator()()’:
> [build] /opt/build/debug/src/test.cxx:57:72: error: left operand of comma operator has no effect [-Werror=unused-value]
> [build]        , "s5"_s + event<e5> [ &self::guard3 ] / (&self::action4, &self::action5)  = X

When combining a callable struct or a lambda with a member function, all is well.
I suppressed the warning, compiled again and executed the test. As expected the action is NOT executed.

Below is a modified version of the example sml/example/actions_guards.cpp.
I modified actions for event5 to execute 2 actions both using a member function ptr, action4 is not executed, where action5 is.

```
//
// Copyright (c) 2016-2020 Kris Jusiak (kris at jusiak dot net)
//
// Distributed under the Boost Software License, Version 1.0.
// (See accompanying file LICENSE_1_0.txt or copy at
// http://www.boost.org/LICENSE_1_0.txt)
//

#include <boost/sml.hpp>
#include <cassert>
#include <iostream>
#include <typeinfo>

namespace sml = boost::sml;

namespace {

struct e1 {};
struct e2 {};
struct e3 {};
struct e4 {};
struct e5 {};

struct actions_guards {
  using self = actions_guards;

  auto operator()() {
    using namespace sml;

    auto guard1 = [] {
      std::cout << "guard1" << std::endl;
      return true;
    };

    auto guard2 = [](int i) {
      assert(42 == i);
      std::cout << "guard2" << std::endl;
      return false;
    };

    auto action1 = [](auto e) { std::cout << "action1: " << typeid(e).name() << std::endl; };

    struct action2 {
      void operator()(int i) {
        assert(42 == i);
        std::cout << "action2" << std::endl;
      }
    };

    // clang-format off
    return make_transition_table(
       *"idle"_s + event<e1> = "s1"_s
      , "s1"_s + event<e2> [ guard1 ] / action1 = "s2"_s
      , "s2"_s + event<e3> [ guard1 && ![] { return false;} ] / (action1, action2{}) = "s3"_s
      , "s3"_s + event<e4> [ !guard1 || guard2 ] / (action1, [] { std::cout << "action3" << std::endl; }) = "s4"_s
      , "s3"_s + event<e4> [ guard1 ] / ([] { std::cout << "action4" << std::endl; }, [this] { action4(); } ) = "s5"_s

     //  2 member function actions
      , "s5"_s + event<e5> [ &self::guard3 ] / (&self::action4, &self::action5)  = X 
    );
    // clang-format on
  }

  bool guard3(int i) const {
    assert(42 == i);
    std::cout << "guard3" << std::endl;
    return true;
  }

  void action4() const { std::cout << "action4 member" << std::endl; }

  void action5(int i, const e5&) {
    assert(42 == i);
    std::cout << "action5" << std::endl;
  }
};
}  // namespace

int main() {
  actions_guards ag{};
  sml::sm<actions_guards> sm{ag, 42};
  sm.process_event(e1{});
  sm.process_event(e2{});
  sm.process_event(e3{});
  sm.process_event(e4{});
  sm.process_event(e5{});
  assert(sm.is(sml::X));
}

// OUTPUT
guard1
action1: N12_GLOBAL__N_12e2E
guard1
action1: N12_GLOBAL__N_12e3E
action2
guard1
guard2
guard1
action4
action4 member
guard3
action5 << Should have action4 member first

```
## Specifications
  - Version: 1.1.3
  - Platform: Linux 
  - Subsystem: RHEL 7.5
  - Compiler: GCC 8.3.1 20190311

--- a/example/actions_guards.cpp
+++ b/example/actions_guards.cpp
@@ -52,7 +52,9 @@ struct actions_guards {
       , "s2"_s + event<e3> [ guard1 && ![] { return false;} ] / (action1, action2{}) = "s3"_s
       , "s3"_s + event<e4> [ !guard1 || guard2 ] / (action1, [] { std::cout << "action3" << std::endl; }) = "s4"_s
       , "s3"_s + event<e4> [ guard1 ] / ([] { std::cout << "action4" << std::endl; }, [this] { action4(); } ) = "s5"_s
-      , "s5"_s + event<e5> [ &self::guard3 ] / &self::action5 = X
+      // Two member-function-pointer actions: wrap() the first so operator, fires correctly.
+      // (&self::action4, &self::action5) would silently run only action5 (built-in comma).
+      , "s5"_s + event<e5> [ &self::guard3 ] / (wrap(&self::action4), &self::action5) = X
     );
     // clang-format on
   }

--- a/include/boost/sml.hpp
+++ b/include/boost/sml.hpp
@@ -2555,6 +2555,13 @@ template <class T1, class T2, __BOOST_SML_REQUIRES(concepts::callable<bool, T1>:
 constexpr auto operator||(const T1 &t1, const T2 &t2) {
   return front::or_<aux::zero_wrapper<T1>, aux::zero_wrapper<T2>>(aux::zero_wrapper<T1>{t1}, aux::zero_wrapper<T2>{t2});
 }
+// Action sequencing: / (a1, a2) or / (a1, a2, a3, ...)
+// At least one operand must be a class type (lambda, functor, or sml::wrap(mfp)) for
+// this user-defined operator, to be selected over the built-in comma operator.
+// Two raw member-function pointers, e.g. / (&C::a, &C::b), are both pointer types
+// (not class types), so the built-in comma is selected instead and only the last
+// action runs.  Wrap the first pointer with sml::wrap() to fix:
+//   / (sml::wrap(&C::a), &C::b)   -- correct: both a and b execute
 template <class T1, class T2, __BOOST_SML_REQUIRES(concepts::callable<void, T1>::value &&concepts::callable<void, T2>::value)>
 constexpr auto operator,(const T1 &t1, const T2 &t2) {
   return front::seq_<aux::zero_wrapper<T1>, aux::zero_wrapper<T2>>(aux::zero_wrapper<T1>{t1}, aux::zero_wrapper<T2>{t2});

--- a/test/ft/transition_table.cpp
+++ b/test/ft/transition_table.cpp
@@ -214,3 +214,34 @@ test transition_table_types = [] {
   float f = 12.0;
   sml::sm<c> sm{f, 42, 87.0, 0.0f};
 };
+
+// Issue #389: two member-function-pointer actions combined with operator,
+// silently ran only the last action because raw MFP types are not class types —
+// the built-in comma operator was selected instead of the user-defined one.
+// Fix: wrap the first MFP with sml::wrap() so it becomes a class type and the
+// overloaded operator, is considered by overload resolution.
+test member_functions_two_mfp_actions_with_wrap = [] {
+  struct c {
+    using self = c;
+
+    auto operator()() noexcept {
+      using namespace sml;
+      // clang-format off
+      return make_transition_table(
+         // wrap() the first MFP so operator, sees a class type and sequences both.
+         *idle + event<e1> / (wrap(&self::action_a), &self::action_b) = X
+      );
+      // clang-format on
+    }
+
+    void action_a() { ++calls; }
+    void action_b() { ++calls; }
+    int calls = 0;
+  };
+
+  c c_{};
+  sml::sm<c> sm{c_};
+  sm.process_event(e1{});
+  expect(sm.is(sml::X));
+  expect(2 == c_.calls);  // both action_a and action_b must have run
+};

